src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1262,6 +1592,31 @@ export class SavedObjectsRepository {
       throw SavedObjectsErrorHelpers.createGenericNotFoundEsUnavailableError(type, id);
     }
 
+    const spacesToEnforce = new Set([SavedObjectsUtils.namespaceIdToString(namespace)]); // Always check/enforce authZ for the active space
+    const existingNamespaces = body?._source?.namespaces || [];
+    const authorizationResult = await this._securityExtension?.checkAuthorization({
```

Comment:
There are a couple of instances where there is a slight difference, but these probably could be adapted to conform without a difference in behavior. I am not sure why the original wrapper functionality was split into two functions.

Reply:
>  I am not sure why the original wrapper functionality was split into two functions.

I don't recall the history on this, either. I _expect_ it was an attempt to consolidate some of the authorization logic as our needs evolved over time, and/or it could have been a holdover from the late 7.x era where we had to support two distinct implementations of the audit logger.

Reply:
Joe does mention in his deep dive on this PR that there are times when a consumer will want to do something other than pass the result from authorize to enforce. We only have really minor instances of that internal to the repo, so I can only assume he means for 3rd party or future use flexibility.

Reply:
Since we have some minor internal usage, I'm fine to leave this as-is for now then. Thanks for checking!